packages/flutter/lib/src/widgets/app.dart

```diff
@@ -321,6 +378,39 @@ class WidgetsApp extends StatefulWidget {
   /// or a [CupertinoPageRoute] should be used for building page transitions.
   final PageRouteFactory pageRouteBuilder;
 
+  /// {@template flutter.widgets.widgetsApp.routeInformationParser}
+  /// A parser to parse the route information into a generic data type for
+  /// configuring the [routerDelegate].
+  /// {@endtemplate}
+  final RouteInformationParser<Object> routeInformationParser;
+
+  /// {@template flutter.widgets.widgetsApp.routerDelegate}
+  /// A delegate that configures a [Navigator] that will be used by the [Router].
+  ///
+  /// The generic type `T` must match the generic type of the
+  /// [routeInformationParser].
+  /// {@endtemplate}
+  final RouterDelegate<Object> routerDelegate;
+
+  /// {@template flutter.widgets.widgetsApp.backButtonDispatcher}
+  /// A back button dispatcher that decide whether to handle the android back
+  /// button intent.
+  ///
+  /// If this is not provided, the widgets app will create a
+  /// [RootBackButtonDispatcher] by default.
+  /// {@endtemplate}
+  final BackButtonDispatcher backButtonDispatcher;
+
+  /// {@template flutter.widgets.widgetsApp.routeInformationProvider}
+  /// A route information provider that provides the route information for the
+  /// [Router].
```

Comment:
this sentence tells me nothing about what this does. :-) it just repeats the same words as in the name of the types involved...